src/States.cpp

```diff
@@ -0,0 +1,472 @@
+#include "States.h"
+
+#include <iostream>
+#include <stdexcept>
+#include <algorithm>
+#include <gsl/gsl_interp.h>
+#include <gsl/gsl_spline.h>
+#include <gsl/gsl_poly.h>
+#include <cmath>
+#include <float.h>
+
+namespace ale {
+  
+  /** The following helper functions are used to calculate the reduced states cache and cubic hermite 
+  to interpolate over it. They were migrated, with minor modifications, from 
+  Isis::NumericalApproximation **/
+  
+  /** Checks to see if an individual value, a, is within the doman specified by x **/
+  bool InsideDomain(const double a, std::vector<double> x) {
+    if(a + DBL_EPSILON < *min_element(x.begin(), x.end())) {
+      return false;
+    }
+    if(a - DBL_EPSILON > *max_element(x.begin(), x.end())) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
Where is DBL_EPSILON defined?

Reply:
It's in the c++ header file:
<cfloat>  or <float.h>

Reply:
Which cpp header file, aren't there a bunch? I also meant what is its value? Does it have some standard value in that header file?

Reply:
For true c++ the header is: cfloat
The C header is: float.h

Reply:
Ohhh that is what you meant, thank you.